pkg/registry/apis/iam/user/rest_display.go

```diff
@@ -1,123 +1,145 @@
 }
 
-var (
-	_ rest.Storage              = (*LegacyDisplayREST)(nil)
-	_ rest.SingularNameProvider = (*LegacyDisplayREST)(nil)
-	_ rest.Connecter            = (*LegacyDisplayREST)(nil)
-	_ rest.Scoper               = (*LegacyDisplayREST)(nil)
-	_ rest.StorageMetadata      = (*LegacyDisplayREST)(nil)
-)
-
 func NewLegacyDisplayREST(store legacy.LegacyIdentityStore) *LegacyDisplayREST {
 	return &LegacyDisplayREST{store}
 }
 
-func (r *LegacyDisplayREST) New() runtime.Object {
-	return &iamv0.DisplayList{}
-}
-
-func (r *LegacyDisplayREST) Destroy() {}
-
-func (r *LegacyDisplayREST) NamespaceScoped() bool {
-	return true
-}
-
-func (r *LegacyDisplayREST) GetSingularName() string {
-	return "display"
-}
-
-func (r *LegacyDisplayREST) ProducesMIMETypes(verb string) []string {
-	return []string{"application/json"}
-}
-
-func (r *LegacyDisplayREST) ProducesObject(verb string) any {
-	return &iamv0.DisplayList{}
-}
-
-func (r *LegacyDisplayREST) ConnectMethods() []string {
-	return []string{http.MethodGet}
-}
-
-func (r *LegacyDisplayREST) NewConnectOptions() (runtime.Object, bool, string) {
-	return nil, false, "" // true means you can use the trailing path as a variable
+func (r *LegacyDisplayREST) GetAPIRoutes(defs map[string]common.OpenAPIDefinition) *builder.APIRoutes {
+	resultSchema := defs["github.com/grafana/grafana/pkg/apis/iam/v0alpha1.DisplayList"].Schema
+	return &builder.APIRoutes{
+		Namespace: []builder.APIRouteHandler{
+			{
+				Path: "display",
+				Spec: &spec3.PathProps{
+					Get: &spec3.Operation{
+						OperationProps: spec3.OperationProps{
+							Tags:        []string{"Display"},
+							Description: "Show user display information",
+							Parameters: []*spec3.Parameter{
+								{
+									ParameterProps: spec3.ParameterProps{
+										Name:        "namespace",
+										In:          "path",
+										Required:    true,
+										Example:     "default",
+										Description: "workspace",
+										Schema:      spec.StringProperty(),
+									},
+								},
+								{
+									ParameterProps: spec3.ParameterProps{
+										Name:        "key",
+										In:          "query",
+										Description: "Display keys",
+										Required:    true,
+										Example:     "user:u000000001",
+										Schema:      spec.ArrayProperty(spec.StringProperty()),
+										//	Style:       "form",
+										Explode: true,
+									},
+								},
+							},
+							Responses: &spec3.Responses{
+								ResponsesProps: spec3.ResponsesProps{
+									StatusCodeResponses: map[int]*spec3.Response{
+										200: {
+											ResponseProps: spec3.ResponseProps{
+												Content: map[string]*spec3.MediaType{
+													"application/json": {
+														MediaTypeProps: spec3.MediaTypeProps{
+															Schema: &resultSchema,
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Handler: r.handleDisplay,
+			},
+		},
+	}
```

Comment:
This is pretty rough :( It would be nice if we could get this part generated for these kind of handlers. How you would do that I am not sure but still, I suspect we will have quite few of raw endpoints within iam

Reply:
> This is pretty rough :(

<img width="100" alt="image" src="https://github.com/user-attachments/assets/b2736110-fdc5-4e35-9b54-b586673a04d1" />

um, ya -- this is the case where we do not have a great model yet 😬   Ideally it is infrequent so not the worst, but for sure something that needs to be improved.